Index: index.php
===================================================================
--- index.php	(revision 167868)
+++ index.php	(working copy)
@@ -50,7 +50,7 @@
 }
 
 $mageFilename = MAGENTO_ROOT . '/app/Mage.php';
-$maintenanceFile = 'maintenance.flag';
+$maintenanceFile = '.htmaint';
 
 if (!file_exists($mageFilename)) {
     if (is_dir('downloader')) {
@@ -62,8 +62,13 @@
 }
 
 if (file_exists($maintenanceFile)) {
-    include_once dirname(__FILE__) . '/errors/503.php';
-    exit;
+    ob_start();
+    $maintenanceDevIps = include $maintenanceFile;
+    ob_end_clean();
+    if (!is_array($maintenanceDevIps) || !in_array($_SERVER['REMOTE_ADDR'], $maintenanceDevIps)) {
+        include_once dirname(__FILE__) . '/errors/503.php';
+        exit;
+    }
 }
 
 require_once $mageFilename;
