Index: lib/Varien/Image/Adapter/Abstract.php
===================================================================
--- lib/Varien/Image/Adapter/Abstract.php	(revision 157611)
+++ lib/Varien/Image/Adapter/Abstract.php	(working copy)
@@ -67,6 +67,13 @@
 
     abstract public function save($destination=null, $newName=null);
 
+    /**
+     * Render image and return its binary contents
+     *
+     * @return string
+     */
+    abstract public function getImage();
+
     abstract public function display();
 
     abstract public function resize($width=null, $height=null);
@@ -81,13 +88,11 @@
 
     public function getMimeType()
     {
-        if( $this->_fileType ) {
-            return $this->_fileType;
-        } else {
+        if( $this->_fileType === null ) {
             list($this->_imageSrcWidth, $this->_imageSrcHeight, $this->_fileType, ) = getimagesize($this->_fileName);
             $this->_fileMimeType = image_type_to_mime_type($this->_fileType);
-            return $this->_fileMimeType;
         }
+        return $this->_fileMimeType;
     }
 
     /**
Index: lib/Varien/Image/Adapter/Gd2.php
===================================================================
--- lib/Varien/Image/Adapter/Gd2.php	(revision 157611)
+++ lib/Varien/Image/Adapter/Gd2.php	(working copy)
@@ -127,6 +127,13 @@
         call_user_func_array($this->_getCallback('output'), $functionParameters);
     }
 
+    public function getImage()
+    {
+        ob_start();
+        call_user_func($this->_getCallback('output'), $this->_imageHandler);
+        return ob_get_clean();
+    }
+
     public function display()
     {
         header("Content-type: ".$this->getMimeType());
