Index: lib/Varien/Data/Form/Element/Multiline.php
===================================================================
--- lib/Varien/Data/Form/Element/Multiline.php	(revision 75180)
+++ lib/Varien/Data/Form/Element/Multiline.php	(working copy)
@@ -63,7 +63,7 @@
                 $this->setClass('input-text');
             }
             $html.= '<div class="multi-input"><input id="'.$this->getHtmlId().$i.'" name="'.$this->getName().'['.$i.']'
-                .'" value="'.$this->getEscapedValue($i).'"'.$this->serialize($this->getHtmlAttributes()).' />'."\n";
+                .'" value="'.$this->getEscapedValue($i).'" '.$this->serialize($this->getHtmlAttributes()).' />'."\n";
             if ($i==0) {
                 $html.= $this->getAfterElementHtml();
             }
