Index: lib/Varien/File/Uploader.php
===================================================================
--- lib/Varien/File/Uploader.php	(revision 75180)
+++ lib/Varien/File/Uploader.php	(working copy)
@@ -136,7 +136,7 @@
     {
         $this->_setUploadFileId($fileId);
         if( !file_exists($this->_file['tmp_name']) ) {
-            throw new Exception('File was not uploaded.');
+            throw new Exception('File was not uploaded.', $this->_file['error']);
             return;
         } else {
             $this->_fileExists = true;
