Index: app/code/core/Mage/Adminhtml/Controller/Action.php
===================================================================
--- app/code/core/Mage/Adminhtml/Controller/Action.php	(revision 110161)
+++ app/code/core/Mage/Adminhtml/Controller/Action.php	(working copy)
@@ -66,6 +66,11 @@
         return true;
     }
 
+    public function isAllowed()
+    {
+        return $this->_isAllowed();
+    }
+
     /**
      * Retrieve adminhtml session model object
      *
