Index: app/code/core/Mage/XmlConnect/Model/Preview/Abstract.php
===================================================================
--- app/code/core/Mage/XmlConnect/Model/Preview/Abstract.php	(revision 106453)
+++ app/code/core/Mage/XmlConnect/Model/Preview/Abstract.php	(working copy)
@@ -220,7 +220,7 @@
 
             $hex = str_replace('#','',$this->getData('conf/categoryItem/tintColor'));
             $hex2 = '';
-            $_rgb = array();
+            $_rgb = array(0, 0, 0);
 
             $d = '[a-fA-F0-9]';
 
