Index: app/code/core/Mage/Connect/controllers/Adminhtml/Extension/CustomController.php
===================================================================
--- app/code/core/Mage/Connect/controllers/Adminhtml/Extension/CustomController.php	(revision 110161)
+++ app/code/core/Mage/Connect/controllers/Adminhtml/Extension/CustomController.php	(working copy)
@@ -43,7 +43,6 @@
              ->_title($this->__('Magento Connect'))
              ->_title($this->__('Package Extensions'));
 
-        Mage::app()->getStore()->setStoreId(1);
         $this->_forward('edit');
     }
 
