Index: index.php
===================================================================
--- index.php	(revision 62939)
+++ index.php	(working copy)
@@ -43,7 +43,7 @@
 }
 
 $mageFilename = 'app/Mage.php';
-$maintenanceFile = 'maintenance.flag';
+$maintenanceFile = '.htmaint';
 
 if (!file_exists($mageFilename)) {
     if (is_dir('downloader')) {
@@ -54,7 +54,7 @@
     exit;
 }
 
-if (file_exists($maintenanceFile)) {
+if (file_exists($maintenanceFile) && !in_array($_SERVER['REMOTE_ADDR'], array('129.93.245.119'))) {
     $basePath = dirname($_SERVER['PHP_SELF']);
     include_once dirname(__FILE__) . '/errors/503.php';
     exit;
Index: js/mage/adminhtml/wysiwyg/tiny_mce/setup.js
===================================================================
--- js/mage/adminhtml/wysiwyg/tiny_mce/setup.js	(revision 62939)
+++ js/mage/adminhtml/wysiwyg/tiny_mce/setup.js	(working copy)
@@ -106,6 +106,8 @@
             theme_advanced_resizing : true,
             convert_urls : false,
             relative_urls : false,
+            body_id : 'maincontent',
+            body_class : 'fixed',
             content_css: this.config.content_css,
             custom_popup_css: this.config.popup_css,
             magentowidget_url: this.config.widget_window_url,
