Index: app/code/core/Mage/Adminhtml/Block/Widget/Grid/Column/Filter/Abstract.php
===================================================================
--- app/code/core/Mage/Adminhtml/Block/Widget/Grid/Column/Filter/Abstract.php	(revision 75180)
+++ app/code/core/Mage/Adminhtml/Block/Widget/Grid/Column/Filter/Abstract.php	(working copy)
@@ -54,7 +54,7 @@
 
     protected function _getHtmlId()
     {
-        return $this->getColumn()->getGrid()->getVarNameFilter().'_'.$this->getColumn()->getId();
+        return $this->getColumn()->getGrid()->getId().'_'.$this->getColumn()->getGrid()->getVarNameFilter().'_'.$this->getColumn()->getId();
     }
 
     public function getEscapedValue($index=null)
