Index: app/code/core/Mage/Adminhtml/Block/Widget/Grid.php
===================================================================
--- app/code/core/Mage/Adminhtml/Block/Widget/Grid.php	(revision 110161)
+++ app/code/core/Mage/Adminhtml/Block/Widget/Grid.php	(working copy)
@@ -596,6 +596,7 @@
         $this->_prepareColumns();
         $this->_prepareMassactionBlock();
         $this->_prepareCollection();
+        Mage::dispatchEvent('adminhtml_block_widget_grid_prepare_grid', array('grid' => $this));
         return $this;
     }
 
