Index: app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php
===================================================================
--- app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php	(revision 107170)
+++ app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php	(working copy)
@@ -696,7 +696,11 @@
         }
 
         // validate table
-        if (strpos($tableName, '/') !== false) {
+        if ($tableName instanceof Varien_Db_Select) {
+            if (empty($tableAlias)) {
+                throw Mage::exception('Mage_Eav', Mage::helper('eav')->__('Missing table alias for joined subquery.'));
+            }
+        } elseif (strpos($tableName, '/') !== false) {
             $tableName = Mage::getSingleton('core/resource')->getTableName($tableName);
         }
         if (empty($tableAlias)) {
