Index: js/varien/product.js
===================================================================
--- js/varien/product.js	(revision 167868)
+++ js/varien/product.js	(working copy)
@@ -582,7 +582,7 @@
 
         this.optionPrices   = {};
         this.customPrices   = {};
-        this.containers     = {};
+        this.containers     = [];
 
         this.displayZeroPrice   = true;
 
@@ -641,14 +641,32 @@
         var optionOldPrice = optionPrices[2];
         var priceInclTax = optionPrices[3];
         optionPrices = optionPrices[0];
+        
+        var subPrice = 0;
+        var subPriceincludeTax = 0;
+        Object.values(this.customPrices).each(function(el){
+            if (el.excludeTax && el.includeTax) {
+                subPrice += parseFloat(el.excludeTax);
+                subPriceincludeTax += parseFloat(el.includeTax);
+            } else {
+                subPrice += parseFloat(el.price);
+                subPriceincludeTax += parseFloat(el.price);
+            }
+        });
+        
+        var priceDiff = optionPrices + subPrice;
+        if (this.lastReloadPrice !== undefined && this.lastReloadPrice === priceDiff) {
+        	return;
+        }
+        this.lastReloadPrice = priceDiff;
 
-        $H(this.containers).each(function(pair) {
+        $A(this.containers).each(function(value) {
             var _productPrice;
             var _plusDisposition;
             var _minusDisposition;
             var _priceInclTax;
-            if ($(pair.value)) {
-                if (pair.value == 'old-price-'+this.productId && this.productOldPrice != this.productPrice) {
+            if ($(value)) {
+                if (value == 'old-price-'+this.productId && this.productOldPrice != this.productPrice) {
                     _productPrice = this.productOldPrice;
                     _plusDisposition = this.oldPlusDisposition;
                     _minusDisposition = this.oldMinusDisposition;
@@ -659,7 +677,7 @@
                 }
                 _priceInclTax = priceInclTax;
 
-                if (pair.value == 'old-price-'+this.productId && optionOldPrice !== undefined) {
+                if (value == 'old-price-'+this.productId && optionOldPrice !== undefined) {
                     price = optionOldPrice+parseFloat(_productPrice);
                 } else if (this.specialTaxPrice == 'true' && this.priceInclTax !== undefined && this.priceExclTax !== undefined) {
                     price = optionPrices+parseFloat(this.priceExclTax);
@@ -683,17 +701,6 @@
                     var incl = excl + tax;
                 }
 
-                var subPrice = 0;
-                var subPriceincludeTax = 0;
-                Object.values(this.customPrices).each(function(el){
-                    if (el.excludeTax && el.includeTax) {
-                        subPrice += parseFloat(el.excludeTax);
-                        subPriceincludeTax += parseFloat(el.includeTax);
-                    } else {
-                        subPrice += parseFloat(el.price);
-                        subPriceincludeTax += parseFloat(el.price);
-                    }
-                });
                 excl += subPrice;
                 incl += subPriceincludeTax;
 
@@ -709,11 +716,11 @@
                 excl += parseFloat(nonTaxable);
                 incl += parseFloat(nonTaxable);
 
-                if (pair.value == 'price-including-tax-'+this.productId) {
+                if (value == 'price-including-tax-'+this.productId) {
                     price = incl;
-                } else if (pair.value == 'price-excluding-tax-'+this.productId) {
+                } else if (value == 'price-excluding-tax-'+this.productId) {
                     price = excl;
-                } else if (pair.value == 'old-price-'+this.productId) {
+                } else if (value == 'old-price-'+this.productId) {
                     if (this.showIncludeTax || this.showBothPrices) {
                         price = incl;
                     } else {
@@ -735,15 +742,15 @@
                     formattedPrice = '';
                 }
 
-                if ($(pair.value).select('.price')[0]) {
-                    $(pair.value).select('.price')[0].innerHTML = formattedPrice;
-                    if ($(pair.value+this.duplicateIdSuffix) && $(pair.value+this.duplicateIdSuffix).select('.price')[0]) {
-                        $(pair.value+this.duplicateIdSuffix).select('.price')[0].innerHTML = formattedPrice;
+                if ($(value).select('.price')[0]) {
+                    $(value).select('.price')[0].innerHTML = formattedPrice;
+                    if ($(value+this.duplicateIdSuffix) && $(value+this.duplicateIdSuffix).select('.price')[0]) {
+                        $(value+this.duplicateIdSuffix).select('.price')[0].innerHTML = formattedPrice;
                     }
                 } else {
-                    $(pair.value).innerHTML = formattedPrice;
-                    if ($(pair.value+this.duplicateIdSuffix)) {
-                        $(pair.value+this.duplicateIdSuffix).innerHTML = formattedPrice;
+                    $(value).innerHTML = formattedPrice;
+                    if ($(value+this.duplicateIdSuffix)) {
+                        $(value+this.duplicateIdSuffix).innerHTML = formattedPrice;
                     }
                 }
             };
@@ -750,25 +757,20 @@
         }.bind(this));
 
         for (var i = 0; i < this.tierPrices.length; i++) {
+        	var tierPrice = 0, tierPriceInclTax = 0;
             $$('.price.tier-' + i).each(function (el) {
-                var price = this.tierPrices[i] + parseFloat(optionPrices);
-                el.innerHTML = this.formatPrice(price);
+                tierPrice = this.tierPrices[i] + optionPrices + subPrice;
+                el.innerHTML = this.formatPrice(tierPrice);
             }, this);
             $$('.price.tier-' + i + '-incl-tax').each(function (el) {
-                var price = this.tierPricesInclTax[i] + parseFloat(optionPrices);
-                el.innerHTML = this.formatPrice(price);
+            	tierPriceInclTax = this.tierPricesInclTax[i] + priceInclTax + subPriceincludeTax;
+                el.innerHTML = this.formatPrice(tierPriceInclTax);
             }, this);
             $$('.benefit').each(function (el) {
-                var parsePrice = function (html) {
-                    return parseFloat(/\d+\.?\d*/.exec(html));
-                };
-                var container = $(this.containers[3]) ? this.containers[3] : this.containers[0];
-                var price = parsePrice($(container).innerHTML);
-                var tierPrice = $$('.tier-price.tier-' + i+' .price');
-                tierPrice = tierPrice.length ? parsePrice(tierPrice[0].innerHTML, 10) : 0;
-                var $percent = Selector.findChildElements(el, ['.percent.tier-' + i]);
-                $percent.each(function (el) {
-                    el.innerHTML = Math.ceil(100 - ((100 / price) * tierPrice));
+            	var showPrice = this.showIncludeTax ? tierPriceInclTax : tierPrice,
+            		percent = Math.ceil(100 - ((100 / price) * showPrice));
+                el.select('.percent.tier-' + i).each(function (el) {
+                    el.innerHTML = percent;
                 });
             }, this);
         }
